src/domain/synthetics/markets/utils.ts

```diff
@@ -439,19 +448,26 @@ export function marketTokenAmountToUsd(marketInfo: MarketInfo, marketToken: Toke
   return convertToUsd(amount, marketToken.decimals, price)!;
 }
 
-export function getContractMarketPrices(tokensData: TokensData, market: Market): ContractMarketPrices | undefined {
+export function getContractMarketPrices(
+  tokensData: TokensData,
+  market: Pick<Partial<Market>, "longTokenAddress" | "shortTokenAddress" | "indexTokenAddress">
+): ContractMarketPrices | undefined {
   const indexToken = getByKey(tokensData, market.indexTokenAddress);
   const longToken = getByKey(tokensData, market.longTokenAddress);
   const shortToken = getByKey(tokensData, market.shortTokenAddress);
 
-  if (!indexToken || !longToken || !shortToken) {
+  if (
+    (market.indexTokenAddress && !indexToken) ||
+    (market.longTokenAddress && !longToken) ||
+    (market.shortTokenAddress && !shortToken)
+  ) {
     return undefined;
   }
 
   return {
-    indexTokenPrice: convertToContractTokenPrices(indexToken.prices, indexToken.decimals),
-    longTokenPrice: convertToContractTokenPrices(longToken.prices, longToken.decimals),
-    shortTokenPrice: convertToContractTokenPrices(shortToken.prices, shortToken.decimals),
+    indexTokenPrice: indexToken && convertToContractTokenPrices(indexToken.prices, indexToken.decimals),
+    longTokenPrice: longToken && convertToContractTokenPrices(longToken.prices, longToken.decimals),
+    shortTokenPrice: shortToken && convertToContractTokenPrices(shortToken.prices, shortToken.decimals),
```

Comment:
what if return undefined if `!indexToken || !longToken || !shortToken`?
why is it changed btw? I see it previously had this check and I think it makes sense it keep it?

Reply:
i guess this part was left form development time and could be reverted as optional values of prices in `ContractMarketPrices` type